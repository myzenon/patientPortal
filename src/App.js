import React, { Component } from 'react'
import Loading from 'components/Loading'
import { ConnectedRouter } from 'react-router-redux'
import Router from './Router'

import { connect } from 'react-redux'
import { authInited } from 'redux/auth/action'
import { push } from 'react-router-redux'
import { resetAuth } from 'redux/auth/action'

// import idle component from react-idle
// reference: https://reacttraining.com/react-idle/
import Idle from 'react-idle'

import '!style-loader!css-loader!bootstrap/dist/css/bootstrap.css'
import 'themes/global.scss'

export class App extends Component {
    componentWillMount() {
        // when the app start then init the auth system.
        this.props.authInited()
    }
    signOut = async () => {
        // reset auth and move to login page
        await this.props.resetAuth()
        this.props.push('/login')
    }
    render() {
        // if api is not initied or no auth in the system.
        if (!this.props.page.apiInited && JSON.stringify(this.props.auth) !== '{}') {
            return (<Loading />)
        }
        return (
            // Using react idle
            <Idle
                // if mouse / keyboard not active within 15 mins
                timeout={900000}
                // on change will call
                onChange={({idle}) => {
                    // when idle and auth is avaliable in the system
                    if (idle && JSON.stringify(this.props.auth) !== '{}') {
                        // then call signout method
                        this.signOut()
                    }
                }}
                render={() => (
                    // render router
                    <ConnectedRouter history={this.props.history}>
                        <Router Loading={this.props.loading ? (<Loading />) : null} />
                    </ConnectedRouter>
                )}
            />
        )
    }
}

export default connect (
    ({
        loading,
        auth,
        page
    }) => ({
        loading,
        auth,
        page
    }),
    {
        authInited,
        push,
        resetAuth
    }
)(App)