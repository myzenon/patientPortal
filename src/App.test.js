import React from 'react'
import createHistory from 'history/createBrowserHistory'
import { App } from './App'
import ReactGA from 'react-ga';

   // This should be part of your setup
   ReactGA.initialize('UA-121783543-1', {
       'debug': true
     });;

jest.mock("react-ga")
jest.mock('assets/policy/nopp.html', () => '<meta charset="UTF-8">\n<meta version="v1.0">\nnopp content')
jest.mock('assets/policy/eula.html', () => '<meta charset="UTF-8">\n<meta version="v1.0">\neula content')
jest.mock('assets/policy/terms.html', () => '<meta charset="UTF-8">\n<meta version="v1.0">\nterms content')
jest.mock('assets/policy/access.html', () => '<meta charset="UTF-8">\n<meta version="v1.0">\naccess content')

test('renders without initial system ', () => {
    const props = {
        authInited: () => {},
        history: createHistory(),
        page: {
            apiInited: false,
            auth: {}
        }
    }
    const wrapper = render (
        <App {...props} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
})