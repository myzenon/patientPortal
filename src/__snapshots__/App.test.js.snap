// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without initial system  1`] = `
<div
  class="wrapper"
>
  <div
    class="loader"
  />
</div>
`;
